modules/apm/src/main/java/org/elasticsearch/telemetry/apm/APMMeterRegistry.java

```diff
@@ -90,6 +92,23 @@ public DoubleCounter getDoubleCounter(String name) {
         return doubleCounters.get(name);
     }
 
+    @Override
+    public DoubleAsyncCounter registerDoubleAsyncCounter(
```

Comment:
How about using a tiny helper to verify the registrar is know using an assertion?
It's easy to forget extending the test case that would catch that. Something like below should help preventing similar copy-paste errors.

```java
    private <T extends AbstractInstrument<?>> T register(Registrar<T> registrar, T adapter){
        try (ReleasableLock lock = registerLock.acquire()) {
            assert registrars.contains(registrar) : "usage of unknown registrar";
            return registrar.register(adapter);
        }
    }
```

Reply:
I like that idea, it would be super useful